### PR TITLE
Tutorial Solutions: Names

### PR DESCRIPTION
**Pull request by ax3l**
Adding names to lambda functions like a good kokkos-citizen :)

### CODE REVIEW

**Comment by ax3l** [comment]
ping @crtrott :)

File: Intro-Full/Exercises/06/Solution/exercise_6_solution.cpp [hunk 1]
```diff
@@ -148,21 +148,21 @@ int main( int argc, char* argv[] )
     // Application: <y,Ax> = y^T*A*x
     double result = 0;
 
-    Kokkos::parallel_reduce( team_policy( E, Kokkos::AUTO, 32 ), KOKKOS_LAMBDA ( const member_type &teamMember, double &update ) {
+    Kokkos::parallel_reduce( "yAx_outer", team_policy( E, Kokkos::AUTO, 32 ), KOKKOS_LAMBDA ( const member_type &teamMember, double &update ) {
       const int e = teamMember.league_rank();
       double tempN = 0;
 
-      Kokkos::parallel_reduce( Kokkos::TeamThreadRange( teamMember, N ), [&] ( const int j, double &innerUpdateN ) {
+      Kokkos::parallel_reduce( "yAx_intermediate", Kokkos::TeamThreadRange( teamMember, N ), [&] ( const int j, double &innerUpdateN ) {
```
Comment (mhoemmen) [inline Intro-Full/Exercises/06/Solution/exercise_6_solution.cpp]: I'm pretty sure it's either impossible or meaningless to label inner `parallel_*` or `single`.

Comment (ax3l) [reply Intro-Full/Exercises/06/Solution/exercise_6_solution.cpp]: You mean it is probably inlined anyway? Did not check actually with hierarchical parallelism. Shall we just not name them?

Comment (mhoemmen) [reply Intro-Full/Exercises/06/Solution/exercise_6_solution.cpp]: @ax3l wrote:
> You mean it is probably inlined anyway?

What I mean is that the string only has an effect for the outermost `parallel_*` function.  Labels  mainly exist for Kokkos profiling, and profiling only happens at the beginning and end of the outermost `parallel_*` function.  However, if this compiles, I'm pretty sure it's harmless to leave the labels in place -- Kokkos could in theory use them for debugging.

Comment (ax3l) [reply Intro-Full/Exercises/06/Solution/exercise_6_solution.cpp]: I removed them anyway, according to @crtrott it has no effect so far and the names I gave were pretty generic anyway.

**Comment by mhoemmen** [comment]
Uh oh, I think I just merged into master instead of develop.  I'll merge into develop too.

**Comment by ax3l** [comment]
Thanks for the great to grasp tutorial!

**Pull request merged**